### PR TITLE
Add alerts

### PR DESCRIPTION
**Pull request by gferraro**

### CODE REVIEW

File: emailUtil.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+): string {
+  const dateTime = moment(recording.recordingDateTime).format(" H:MMa Do MMM");
+  let html = `<h1>${camera} has detected a ${tag.what} - ${dateTime}</h1>`;
+  html += `<a  href="${config.server.recording_url_base}/${recording.id}?device=${recording.DeviceId}">View Recording</a>`;
+  html += "<br><p>Thanks,<br> Cacophony Team</p>";
+  return html;
+}
+
+async function sendEmail(
+  html: string,
```
Comment (hardiesoft) [inline emailUtil.ts:21]: Do we need a text only representation of the email too, or does nodemailer generate one automatically from the html version?

File: models/Alert.ts [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+  automatic: boolean;
+}
+export function isAlertCondition(condition: any) {
+  return condition.hasOwnProperty('tag') && condition.hasOwnProperty('automatic');
+}
+
+export interface Alert extends Sequelize.Model, ModelCommon<Alert> {
+  id: AlertId;
+  UserId: UserId;
+  conditions: any;
```
Comment (hardiesoft) [inline models/Alert.ts:42]: Conditions should be `AlertCondition[]` rather than `any`, right?

**Review by hardiesoft — CHANGES_REQUESTED** [review]

File: models/Event.ts [hunk 3]
```diff
@@ -32,6 +32,16 @@ export interface Event extends Sequelize.Model, ModelCommon<Event> {
...
   Device: Device | null;
+
+  create: ({
+    deviceId,
+    eventDetailId,
+    dateTime
+  }: {
+    deviceId: number;
+    eventDetailId: number;
+    dateTime: Date;
```
[the remark this answers is not in the document]
Comment (gferraro) [reply models/Event.ts:43]: Chanted to pass in a date

File: report-service-errors.ts [hunk 4]
```diff
@@ -26,13 +26,54 @@ async function main() {
...
     "Service Errors in the last 24 hours"
   );
 }
 
+function generateText(
+  startDate: Moment,
+  endDate: Moment,
+  serviceErrors: ServiceErrorMap
+): string {
+  let html = `Service Errors ${startDate.format(
```
Comment (hardiesoft) [inline report-service-errors.ts:44]: Maybe rename these `html` variables?

**Review by hardiesoft — APPROVED** [review]
Looks good to me, thanks for all the work on this!

**Pull request merged**